lib/Onyx.js

```diff
@@ -1248,19 +1259,34 @@ function mergeCollection(collectionKey, collection) {
 
     return getAllKeys()
         .then((persistedKeys) => {
-            // Split to keys that exist in storage and keys that don't
-            const [existingKeys, newKeys] = _.chain(collection)
-                .keys()
-                .partition(key => persistedKeys.includes(key))
-                .value();
+            const existingKeysToKeep = [];
+            const keysToRemove = [];
+            const collectionWithoutKeysToRemove = collection;
+            const keyValuePairsForExistingCollection = [];
+            const keyValuePairsForNewCollection = [];
+            _.each(collection, (value, key) => {
+                if (value === null) {
+                    keysToRemove.push(key);
+                    delete collectionWithoutKeysToRemove[key];
+                    return;
+                }
 
-            const existingKeyCollection = _.pick(collection, existingKeys);
-            const newCollection = _.pick(collection, newKeys);
-            const keyValuePairsForExistingCollection = prepareKeyValuePairsForStorage(existingKeyCollection);
-            const keyValuePairsForNewCollection = prepareKeyValuePairsForStorage(newCollection);
+                if (persistedKeys.includes(key)) {
+                    existingKeysToKeep.push(key);
+                    keyValuePairsForExistingCollection.push([key, removeNullObjectValues(value)]);
```

Comment:
Hmm using `removeNullObjectValues()` seems like a bug here. Wouldn't that mean that when we merge an object and properties that are set to `null` (presumably to remove those top-level values) they will keep existing?

Reply:
Different comment, same line - can we just have `existingKeysToKeep.push(key);` in this if, and then use the `keyValuePairsForExistingCollection.push` on line 1280, instead of doubling it here?

Assuming this is still what we want to do, after @marcaaron's comment